extensions/standard-processors/processors/PutFile.cpp

```diff
@@ -218,11 +218,24 @@ bool PutFile::putFile(core::ProcessSession *session, std::shared_ptr<FlowFileRec
     }
   }
 
-  ReadCallback cb(tmpFile, destFile);
-  session->read(flowFile, &cb);
+  bool success = false;
+
+  if (flowFile->getSize() > 0) {
```

Comment:
In case there is no content of the flowfile, processsession::read is going to rollback:
```
    std::shared_ptr<io::BaseStream> stream = process_context_->getContentRepository()->read(claim);

    if (nullptr == stream) {
      rollback();
      return;
    }
```

Which is fine in a way as you are trying to read something that doesn't exist.